src/sentry/notifications/manager.py

```diff
@@ -0,0 +1,311 @@
+
+
+def _get_target(user_id=None, team_id=None):
+    """
+    Figure out the target from parameters and return it as a tuple.
+
+    :return: (int, int): (target_type, target_identifier)
+    """
+
+    if user_id:
+        return NotificationTargetType.USER.value, user_id
+
+    if team_id:
+        return NotificationTargetType.TEAM.value, team_id
+
+    raise Exception("target must be either a user or a team")
+
+
+def _get_legacy_key(type: NotificationSettingTypes):
+    """
+    Temporary mapping from new enum types to legacy strings.
+
+    :param type: NotificationSettingTypes enum
+    :return: String
+    """
+
+    return KEYS_TO_LEGACY_KEYS.get(type)
+
+
+def _get_legacy_value(type: NotificationSettingTypes, value: NotificationSettingOptionValues):
+    """
+    Temporary mapping from new enum types to legacy strings. Each type has a separate mapping.
+
+    :param type: NotificationSettingTypes enum
+    :param value: NotificationSettingOptionValues enum
+    :return: String
+    """
+
+    return str(KEY_VALUE_TO_LEGACY_VALUE.get(type, {}).get(value))
+
+
+class NotificationsManager(BaseManager):
+    """
+    TODO add a caching layer for notification settings
+    """
+
+    @staticmethod
+    def notify(
+        provider: ExternalProviders,
+        type: NotificationSettingTypes,
+        user_id=None,
+        team_id=None,
+        data=None,
+    ):
+        """
+        Something noteworthy has happened. Let the targets know about what
+        happened on their own terms. For each target, check their notification
+        preferences and send them a message (or potentially do nothing if this
+        kind of correspondence is muted.)
+
+        :param provider: ExternalProviders enum
+        :param type: NotificationSettingTypes enum
+        :param user_id: User object's ID
+        :param team_id: Team object's ID
+        :param data: TODO describe the payload schema
+        :return:
+        """
+        pass
+
+    def get_settings(
+        self,
+        provider: ExternalProviders,
+        type: NotificationSettingTypes,
+        user=None,
+        team=None,
+        project=None,
+        organization=None,
+    ):
+        """
+        In this temporary implementation, always read EMAIL settings from UserOptions.
+
+        :param provider: ExternalProviders enum
+        :param type: NotificationSetting.type enum
+        :param user: TODO
+        :param team: TODO
+        :param project: TODO
+        :param organization: TODO
+
+        :return: NotificationSettingOptionValues enum
+        """
+
+        user_id_option = getattr(user, "id", None)
+        team_id_option = getattr(team, "id", None)
+        scope_type, scope_identifier = _get_scope(
+            user_id_option, project=project, organization=organization
+        )
+        target_type, target_identifier = _get_target(user_id_option, team_id_option)
+
+        _value = (
```

Comment:
I'm a little confused about this since it's not used - is it meant to be used in the TODO validation?